rasa/core/brokers/kafka.py

```diff
@@ -137,15 +144,21 @@ def _create_producer(self) -> None:
                 ssl_check_hostname=False,
             )
         elif self.security_protocol == "SASL_PLAINTEXT":
-            self.producer = kafka.KafkaProducer(
-                client_id=self.client_id,
-                bootstrap_servers=self.url,
-                value_serializer=lambda v: json.dumps(v).encode(DEFAULT_ENCODING),
-                sasl_plain_username=self.sasl_username,
-                sasl_plain_password=self.sasl_password,
-                sasl_mechanism="PLAIN",
-                security_protocol=self.security_protocol,
-            )
+            try:
```

Comment:
how about moving the producer creation to a method so that we only need one `try ... except` block?

Reply:
Since the constructor params change depending on the security protocol, I did a slight re-factor so that it's passed in as a dict of arguments into one try-except block where the KafkaProducer is instantiated. Does this approach make sense or is there a better way to handle it?